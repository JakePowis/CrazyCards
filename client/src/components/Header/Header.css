
.header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: white;
    min-height: 4rem;
    border-bottom: 2px solid rgba(0,0,0, 0.2);
    padding: 0rem 12vw;

  }
  
  .logoImage {
    max-height: 6rem;
    width: auto;
    border: solid white 1px;
    border-radius: 4px;

  }
  
  .logoImage:hover {
    border: solid navy 1px;
  }
  
  .menu {
    color: #5da0f3;
    font-size: 3rem;
 
  }
  
  .menu:hover {
    color: navy
  }
  