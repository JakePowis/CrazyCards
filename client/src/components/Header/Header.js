import React, {useState} from 'react'
import { Link } from 'react-router-dom';
import logo from './crazyCards.png'
import './Header.css'

export default function Header({setToggleView}) {

const [sidebarOpen, setSidebarOpen] = useState(false)

    const openMenu = () => {
        if (!sidebarOpen) {
        setSidebarOpen(true)
        document.querySelector(".sidebar").classList.add("open");
        }
        else {
        setSidebarOpen(false)
        document.querySelector(".sidebar").classList.remove("open")
        }
      }

    return (
     
    <header className="header">
        <Link to='/'><div className="logo" onClick={()=>setToggleView("welcome")}><img src={logo} className="logoImage" alt="crazy cards logo"/></div></Link>
        <div data-testid="menu" className="menu" onClick={openMenu}><i className="fa fa-bars" aria-hidden="true" > </i></div>
    </header>

    )
}
