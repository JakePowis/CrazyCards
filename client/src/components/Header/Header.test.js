import React from 'react';
import ReactDom from 'react-dom';
import Header from './Header.js'
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from "react-test-renderer"

afterEach(cleanup)

test("Header Renders Correctly", ()=> {
    const div = document.createElement("div");
    ReactDom.render( <Router> <Header /></Router>, div)
})

test('Renders menu', () => {
    const { getByTestId } = render(
        <Router>
            <Header />
        </Router>);
    const divElement = getByTestId("menu");
    expect(divElement).toBeInTheDocument();
  });

  test("matches snapshot", ()=> {
    const tree = renderer.create(<Router> <Header /></Router>).toJSON()
    expect(tree).toMatchSnapshot();
})
  