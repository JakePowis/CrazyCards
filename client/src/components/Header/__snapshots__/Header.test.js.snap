// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  " ",
  <header
    className="header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div
        className="logo"
        onClick={[Function]}
      >
        <img
          alt="crazy cards logo"
          className="logoImage"
          src="crazyCards.png"
        />
      </div>
    </a>
    <div
      className="menu"
      data-testid="menu"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-bars"
      >
         
      </i>
    </div>
  </header>,
]
`;
